libbeat/mapping/field.go

```diff
@@ -110,6 +110,9 @@ func (d *DynamicType) Unpack(s string) error {
 
 // Validate ensures objectTypeParams are not mixed with top level objectType configuration
 func (f *Field) Validate() error {
+	if err := f.validateType(); err != nil {
+		return err
+	}
 	if len(f.ObjectTypeParams) == 0 {
 		return nil
```

Comment:
Is it possible that this returns an "blank screen error"? It will be awesome if you can just improve it a bit in this PR too, please please please :)

Reply:
I have reordered the code here so error conditions are checked first and `return nil` is only at the end.
We could consider moving this code to `validateType()`, because these `ObjectTypeParams` should only be set for `objects`, but lets leave this for a follow up if needed.